spec/nsqlookup.yml

```diff
@@ -0,0 +1,448 @@
+        # 404:
+          # description: Not found
+          # schema:
+            # type: object
+            # properties:
+              # message:
+                # description: Specific error condition
+                # type: string
+                # default: TOPIC_NOT_FOUND
+                # enum:
+                  # - TOPIC_NOT_FOUND
+
+  /nodes:
+    get:
+      operationId: doNodes
+      summary: Returns a list of all known nsqd
+      description: |
+        Returns a list of all known nsqd
+      tags:
+        - nsqlookupd
+      responses:
+        200:
+          description: List of nodes
+          schema:
+            type: object
+            properties:
+              producers:
+                type: array
+                items:
+                  $ref: '#/definitions/Node'
+
+  # TODO: Alternate path based on REST
+  # /topics/{topic}:
+  /topic/create:
+    post:
+      operationId: doCreateTopic
+      summary: Creates a topic
+      description: |
+        Creates a topic
+      tags:
+        - nsqlookupd
+      parameters:
+        - name: topic
+          in: query
+          description: the name of topic to create
+          type: string
+          pattern: '^[\.a-zA-Z0-9_-]+(#ephemeral)?$'
+          minLength: 1
+          maxLength: 64
+          required: true
+      responses:
+        # TODO: Consider 201 or 204
+        200:
+          description: Success
+        # TODO: Consider TOPIC_ALREADY_EXISTS
+        400:
+          description: Bad request
+          schema:
+            type: object
+            properties:
+              message:
+                description: Specific error condition
+                type: string
+                default: INVALID_REQUEST
+                enum:
+                  - INVALID_REQUEST
+                  - MISSING_ARG_TOPIC
+                  - INVALID_ARG_TOPIC
+
+  # TODO: Alternate path based on REST
+  # /topics/{topic}:
+  #   delete:
+  /topic/delete:
+    # TODO: Consider 'delete' vs. 'post'
+    post:
+      operationId: doDeleteTopic
+      summary: Deletes a topic
+      description: |
+        Deletes a topic and any channels defined under the specific topic
+      tags:
+        - nsqlookupd
+      parameters:
+        - name: topic
+          in: query
+          description: the name of topic to delete
+          type: string
+          required: true
+      responses:
+        # TODO: Consider 201 or 204
+        200:
+          description: Success
+        400:
+          description: Bad request
+          schema:
+            type: object
+            properties:
+              message:
+                description: Specific error condition
+                type: string
+                default: INVALID_REQUEST
+                enum:
+                  - INVALID_REQUEST
+                  - MISSING_ARG_TOPIC
+        # TODO: 404 TOPIC_NOT_FOUND
+
+  # TODO: Alternate path based on REST
+  # /topics/{topic}/nodes/{node}:
+  #   delete:

```

Comment:
Does everyone agree on these TODOs? It might be best to keep them out of the spec for now.

I'll also put in my 2c now: I prefer RPC style paths to REST style paths. Treating paths as objects is usually a fantasy that needs to be worked-around, and is also less convenient and less efficient for parsing and routing in other tools (e.g. nginx, grep).

I recently-ish worked on a big-ish team project which involved a "REST API" app, and that's my experience. Ended up a lot messier than just using params for params.